deploy/Datadog.Trace.ClrProfiler.WindowsInstaller/Product.wxs

```diff
@@ -50,6 +50,30 @@
       <ComponentGroupRef Id="Files.Native.32"/>
       <?endif ?>
     </Feature>
+
+    <!-- Use Quiet Execution Custom Action which will redirect standard out to the MSI log (see: https://wixtoolset.org/documentation/manual/v3/customactions/qtexec.html) -->
+    <!-- Use SetProperty immediately before calling the script so we can inject the INSTALLFOLDER path -->
+    <SetProperty Id="CA_QtExec_AddDatadogHttpModule"
+                 Before="CA_QtExec_AddDatadogHttpModule"
+                 Sequence="execute"
+                 Value="&quot;cmd&quot; /c &quot;[INSTALLFOLDER]install.cmd&quot;" />
+    <CustomAction Id="CA_QtExec_AddDatadogHttpModule" BinaryKey="WixCA" DllEntry="WixQuietExec"
+                  Execute="deferred" Return="check" Impersonate="no" />
+
+    <!-- Use SetProperty immediately before calling the script so we can inject the INSTALLFOLDER path -->
+    <SetProperty Id="CA_QtExec_RemoveDatadogHttpModule"
+                 Before="CA_QtExec_RemoveDatadogHttpModule"
+                 Sequence="execute"
+                 Value="&quot;cmd&quot; /c &quot;[INSTALLFOLDER]uninstall.cmd&quot;" />
+    <CustomAction Id="CA_QtExec_RemoveDatadogHttpModule" BinaryKey="WixCA" DllEntry="WixQuietExec"
+                  Execute="deferred" Return="check" Impersonate="no" />
+
+    <!-- Call above Custom Actions at the appropriate times during installation (see: https://stackoverflow.com/a/17608049)-->
+    <!-- The idea is to run the install CA only on new installs and the uninstall CA only on uninstalls and upgrades -->
+    <InstallExecuteSequence>
+      <Custom Action="CA_QtExec_AddDatadogHttpModule" After="InstallFiles">NOT Installed AND NOT UPGRADINGPRODUCTCODE</Custom>
```

Comment:
@zacharycmontoya Can you explain where the `UPGRADINGPRODUCTCODE` symbol is defined?

Reply:
The official reference seems to be here with the key text being: `The UPGRADINGPRODUCTCODE property is set by Windows Installer when an upgrade removes an application.`
https://docs.microsoft.com/en-us/windows/win32/msi/upgradingproductcode

Reply:
I'm not sure the `UPGRADINGPRODUCTCODE` condition is really necessary for this, but I noticed that the latest Application Insights Monitor uses this so I figured it would work out

Reply:
TIL Thanks!